controller/src/lib.rs

```diff
@@ -1,86 +1,85 @@
-                        tracing::info!(spawn_request=?schedule_request.value, "Got spawn request");
-                        let result = match scheduler.schedule(&schedule_request.value.cluster, Utc::now()) {
-                            Ok(drone_id) => {
-                                let timer = Timer::new();
-                                let spawn_request = schedule_request.value.schedule(&drone_id);
-                                match nats.request(&spawn_request).await {
-                                    Ok(true) => {
-                                        tracing::info!(
-                                            duration=?timer.duration(),
-                                            backend_id=%spawn_request.backend_id,
-                                            %drone_id,
-                                            "Drone accepted backend."
-                                        );
-                                        ScheduleResponse::Scheduled {
-                                            drone: drone_id,
-                                            backend_id: spawn_request.backend_id,
-                                            bearer_token: spawn_request.bearer_token,
-                                        }
-                                    }
-                                    Ok(false) => {
-                                        tracing::warn!("No drone available.");
-                                        ScheduleResponse::NoDroneAvailable
-                                    },
-                                    Err(error) => {
-                                        tracing::warn!(?error, "Scheduler returned error.");
-                                        ScheduleResponse::NoDroneAvailable
-                                    },
-                                }
-                            },
-                            Err(error) => {
-                                tracing::warn!(?error, "Communication error during scheduling.");
-                                ScheduleResponse::NoDroneAvailable
-                            },
-                        };
-
-                        schedule_request.respond(&result).await?;
-                    },
-                    None => return Err(anyhow!("spawn_request_sub.next() returned None.")),
+                        ScheduleResponse::Scheduled {
+                            drone: drone_id,
+                            backend_id: spawn_request.backend_id,
+                            bearer_token: spawn_request.bearer_token.clone(),
+                        }
+                    }
+                    Ok(false) => {
+                        tracing::warn!("No drone available.");
+                        ScheduleResponse::NoDroneAvailable
+                    }
+                    Err(error) => {
+                        tracing::warn!(?error, "Scheduler returned error.");
+                        ScheduleResponse::NoDroneAvailable
```

Comment:
Should we have a `ScheduleResponse::InternalError` or some other error type here? Or does this error really mean that no drone is available?

Reply:
Currently `NoDroneAvailable` is a catch-all for instances that don't spawn. I'm trying to minimize the surface area of API changes in this PR, so I've opened #335 to track this as a follow-up.